include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -560,6 +564,17 @@ struct __usm_host_or_buffer_storage
     }
 };
 
+template <typename _Acc>
+auto
+__get_usm_host_or_buffer_accessor_ptr(const _Acc& __acc)
```

Comment:
I would suggest to encapsulate this global function to `struct __usm_host_or_buffer_storage` scope (as a static method f.e)

Reply:
I've made this change. One drawback of this I see is that it convolutes the call a bit.